lib/generateClass.js

```diff
@@ -350,12 +581,25 @@ function writeToFhir(def, fhir, cw) {
             // Generate a chain if mapped multiple levels deep, such as myField1.subField2
             const classMethodChain = fieldMapPath.map(e => toSymbol(fhirMappingToIdentifier(e).name)).join('.');
             // find related field if available
-            const field = def.fields.find(f => f.identifier && f.identifier.equals(fhirMappingToIdentifier(fieldMapPath[0])));
-            if(field !== undefined){
+            let field;
+            let currDef = def;
+
+            // for nested element mappings, such as "Patient.person.address", we have to look up Patient.person and then Person.address
+            for(const pathElement of fieldMapPath) {
+              field = currDef.fields.find(f => f.identifier && f.identifier.equals(fhirMappingToIdentifier(pathElement)));
+
+              if (!field) {
+                break;
+              }
+
+              currDef = specs.dataElements.byNamespace(field.identifier.namespace).find(e =>  e.identifier.equals(field.identifier));
+            }
```

Comment:
This block (lines 584 - 596) looks very similar to code I commented on up above.  If so, consider breaking it out into a re-usable function -- especially since I think I suggested changes to be made in the logic.